In [6]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

%matplotlib inline

import warnings
warnings.filterwarnings('ignore')

```%matplotlib inline```是一个特殊的jupyter notebook的魔术命令，它用于在jupyter notebook中显示图形输出。它的作用是将生成的图形嵌入到notebook中，而不是在外部窗口中显示。

In [7]:
train = pd.read_csv('../Titanic/train.csv')
test = pd.read_csv('../Titanic/test.csv')


In [8]:
print(train.columns)

Index(['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp',
       'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked'],
      dtype='object')


In [9]:
train.sample(10)

PassengerId  Survived  Pclass  \
124          125         0       1   
602          603         0       1   
680          681         0       3   
387          388         1       2   
138          139         0       3   
577          578         1       1   
417          418         1       2   
161          162         1       2   
140          141         0       3   
419          420         0       3   

                                                  Name     Sex   Age  SibSp  \
124                        White, Mr. Percival Wayland    male  54.0      0   
602                          Harrington, Mr. Charles H    male   NaN      0   
680                                Peters, Miss. Katie  female   NaN      0   
387                                   Buss, Miss. Kate  female  36.0      0   
138                                Osen, Mr. Olaf Elon    male  16.0      0   
577          Silvey, Mrs. William Baird (Alice Munger)  female  39.0      1   
417                      Silven, Miss. Lyyli Karoliina  female  18.0      0   
161  Watt, Mrs. James (Elizabeth "Bessie" Inglis Mi...  female  40.0      0   
140                      Boulos, Mrs. Joseph (Sultana)  female   NaN      0   
419                          Van Impe, Miss. Catharina  female  10.0      0   

     Parch      Ticket     Fare Cabin Embarked  
124      1       35281  77.2875   D26        S  
602      0      113796  42.4000   NaN        S  
680      0      330935   8.1375   NaN        Q  
387      0       27849  13.0000   NaN        S  
138      0        7534   9.2167   NaN        S  
577      0       13507  55.9000   E44        S  
417      2      250652  13.0000   NaN        S  
161      0  C.A. 33595  15.7500   NaN        S  
140      2        2678  15.2458   NaN        C  
419      2      345773  24.1500   NaN        S

In [10]:
train.describe(include='all')

PassengerId    Survived      Pclass                     Name   Sex  \
count    891.000000  891.000000  891.000000                      891   891   
unique          NaN         NaN         NaN                      891     2   
top             NaN         NaN         NaN  Braund, Mr. Owen Harris  male   
freq            NaN         NaN         NaN                        1   577   
mean     446.000000    0.383838    2.308642                      NaN   NaN   
std      257.353842    0.486592    0.836071                      NaN   NaN   
min        1.000000    0.000000    1.000000                      NaN   NaN   
25%      223.500000    0.000000    2.000000                      NaN   NaN   
50%      446.000000    0.000000    3.000000                      NaN   NaN   
75%      668.500000    1.000000    3.000000                      NaN   NaN   
max      891.000000    1.000000    3.000000                      NaN   NaN   

               Age       SibSp       Parch  Ticket        Fare    Cabin  \
count   714.000000  891.000000  891.000000     891  891.000000      204   
unique         NaN         NaN         NaN     681         NaN      147   
top            NaN         NaN         NaN  347082         NaN  B96 B98   
freq           NaN         NaN         NaN       7         NaN        4   
mean     29.699118    0.523008    0.381594     NaN   32.204208      NaN   
std      14.526497    1.102743    0.806057     NaN   49.693429      NaN   
min       0.420000    0.000000    0.000000     NaN    0.000000      NaN   
25%      20.125000    0.000000    0.000000     NaN    7.910400      NaN   
50%      28.000000    0.000000    0.000000     NaN   14.454200      NaN   
75%      38.000000    1.000000    0.000000     NaN   31.000000      NaN   
max      80.000000    8.000000    6.000000     NaN  512.329200      NaN   

       Embarked  
count       889  
unique        3  
top           S  
freq        644  
mean        NaN  
std         NaN  
min         NaN  
25%         NaN  
50%         NaN  
75%         NaN  
max         NaN

观察数据
1. 总乘员 891 名
2. label 存活
3. 无缺失feature ：Pclass name sex SibSp Parch Ticket Fare 
4. 少量确实feature : Age Embarked
5. 大量确实feature  : cabin


In [11]:
print(pd.isnull(train).sum())

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64
